### PR TITLE
Updates to `report` CLI. Closes #1390

### LINKED ISSUE #1390: `report` CLI updates
- [x]  `ctrl+c` should gracefully exit `steampipe report`
- [x]  `--reportserver-port <1-65535>` allows the webserver to bind to the specified in the specified port
- [x]  `--reportserver-listen <local | network>` - `local` to listen on `localhost` & `network` to listen on `*`

### PR DESCRIPTION
**Pull request by binaek**

### CODE REVIEW

File: report/reportserver/server.go [hunk 1]
```diff
@@ -113,10 +138,7 @@ func buildLeafNodeProgressPayload(event *reportevents.LeafNodeProgress) []byte {
 	//jsonString, _ := json.Marshal(payload)
 	//return jsonString
 	jsonString, err := json.MarshalIndent(payload, "", "  ")
-	fmt.Println(err)
-	a := string(jsonString)
-	fmt.Println(a)
-
+	log.Println("[ERROR] buildLeafNodeProgressPayload marshal error:", err)
```
Comment (kaidaguerre) [inline report/reportserver/server.go:141]: do we really want to print that out?

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

File: report/reportserver/api.go [hunk 2]
```diff
@@ -3,17 +3,16 @@ package reportserver
...
-	"os"
 	"os/exec"
-	"os/signal"
 	"path"
 	"runtime"
 	"time"
 
+	"github.com/spf13/viper"
+	"github.com/turbot/steampipe/constants"
+
```
[the remark this answers is not in the document]
Comment (binaek) [reply report/reportserver/api.go]: done

File: report/reportserver/server.go (hunk 1 above)
Comment (binaek) [reply report/reportserver/server.go]: removed

File: report/reportserver/server.go [hunk 3]
```diff
@@ -109,7 +108,10 @@ func NewServer(ctx context.Context) (*Server, error) {
 	return server, err
 }
 
-func buildAvailableReportsPayload(reports map[string]*modconfig.ReportContainer) []byte {
+// we don't expect the build functions to ever error during marshalling
```
Comment (kaidaguerre) [inline report/reportserver/server.go:111]: move this comment to the calling code where the panic is

File: report/reportserver/server.go [hunk 4]
```diff
@@ -239,12 +229,18 @@ func (s *Server) HandleWorkspaceUpdate(event reportevents.ReportEvent) {
 
 	case *reportevents.WorkspaceError:
 		log.Println("[TRACE] Got workspace error event", *e)
-		payload := buildWorkspaceErrorPayload(e)
+		payload, err := buildWorkspaceErrorPayload(e)
+		if err != nil {
```
Comment (kaidaguerre) [inline report/reportserver/server.go:233]: can we put the panic into a defer block so we only call it once

Comment (kaidaguerre) [reply report/reportserver/server.go:233]: i.e. set an `err` variable and break instead of panicking - then check `err` in the defer

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

**Pull request merged**